spring-integration-hazelcast/src/main/java/org/springframework/integration/hazelcast/leader/LeaderInitiator.java

```diff
@@ -252,6 +276,18 @@ public void destroy() {
 		stop();
 		this.executorService.shutdown();
 	}
+	
+	
+     FencedLock getLock() {
+		CPSubsystem cpSubSystem = client.getCPSubsystem();
+		FencedLock lock = cpSubSystem.getLock(this.candidate.getRole());
```

Comment:
So, your point is to get a fresh lock instance every time we would like to acquire it?

Reply:
Yes, or rather get the fresh lock instance every time you want to interact with the lock. according to my experience, caching local refs is not a good practice when it comes to Hazelcast data structures. However the issues I have seen was related to the HZ Failover client (Disaster/Recovery client mode) and when a cluster switch between datacenters happened. But as a practice I always strive to get the data structures from the client every time I want to use them,